generators/server/templates/src/main/java/package/web/rest/util/_PaginationUtil.java

```diff
@@ -63,7 +71,13 @@ private static String generateUri(String baseUrl, int page, int size) {
     <%_ if (searchEngine === 'elasticsearch') { _%>
 
     public static HttpHeaders generateSearchPaginationHttpHeaders(String query, Page page, String baseUrl) {
-        String escapedQuery = query.replace(",", "%2C");
+        String escapedQuery = query;
+        try {
+            escapedQuery = URLEncoder.encode(query, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            log.info("Unsupported encoding exception");
+            log.trace("Unsupported encoding exception trace: {}", e);
```

Comment:
Just throw an exception, wrapped into a runtime exception, there is no reason to do anything, if UTF-8 is not supported by VM.
And you can make the escapedQuery a final variable - this double assignment is not necessary at all